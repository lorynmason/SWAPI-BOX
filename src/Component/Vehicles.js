import React from 'react'
import './styles/main.scss'
import PropTypes from 'prop-types'

const Vehicles = ({ vehicles, addFavorites }) => {
  const cards = vehicles.map((vehicle) => {
    return (
      <div className="vehicle-card card">
        <button className="favorite-btn" type="button" onClick={() => addFavorites(vehicle.id)}>
          <i className="fas fa-jedi" />
        </button>
        <h1>{vehicle.name}</h1>
        <p>
          {`Model: `}
          {vehicle.model}
        </p>
        <p>
          {`Class: `}
          {vehicle.class}
        </p>
        <p>
          {`Passengers: `}
          {vehicle.passengers}
        </p>
      </div>)
  })

  return (
    <section className="cardContainer">
      {cards}
    </section>
  )
}

Vehicles.propTypes = {
  vehicles: PropTypes.array.isRequired
}

export default Vehicles
