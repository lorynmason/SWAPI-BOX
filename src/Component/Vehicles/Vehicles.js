import React from 'react'
import '../styles/main.scss'
import PropTypes from 'prop-types'

const Vehicles = ({ vehicles, toggleFavorites }) => {
  const cards = vehicles.map((vehicle) => {
    const infoValue = Object.values(vehicle.info).map((vhInfo) => {
      return(
        <p>{vhInfo}</p>
        )
    })

    return (
      <div className="vehicle-card card">
        <button className="favorite-btn" type="button" onClick={() => toggleFavorites(vehicle.id)}>
          <i className="fas fa-jedi" />
        </button>
        <div className='info'>
          <p>
            {infoValue}
          </p>
        </div>
      </div>
      )
  })

  return (
    <section className="cardContainer">
      {cards}
    </section>
  )
}

Vehicles.propTypes = {
  vehicles: PropTypes.array.isRequired
}

export default Vehicles
