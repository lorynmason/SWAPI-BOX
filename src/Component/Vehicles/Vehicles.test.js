import React from 'react'
import { shallow } from 'enzyme'
import Vehicles from './Vehicles';

describe('Vehicles', () => {
  const toggleFavorites = jest.fn()
  let wrapper;
  let mockVehicleData = [
    {
      info: {
        Class:"Vehicle:  wheeled",
        Model:"Model:  Digger Crawler",
        Name:"Sand Crawler",
        Passengers:"Passengers:  30"
      }
    }
  ]
  beforeEach(() => {
    wrapper = shallow(<Vehicles vehicles={mockVehicleData} toggleFavorites={toggleFavorites}/>)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('favorite button should call toggleFavorites on click', () => {
    const button = wrapper.find('button')
    button.simulate('click')

    expect(toggleFavorites).toHaveBeenCalled()
  })
})