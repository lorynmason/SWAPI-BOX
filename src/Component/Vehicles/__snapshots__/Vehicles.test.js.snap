// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vehicles should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Vehicles
    toggleFavorites={[MockFunction]}
    vehicles={
      Array [
        Object {
          "info": Object {
            "Class": "Vehicle:  wheeled",
            "Model": "Model:  Digger Crawler",
            "Name": "Sand Crawler",
            "Passengers": "Passengers:  30",
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="vehicle-card card"
          id="no"
        >
          <button
            className="favorite-btn"
            onClick={[Function]}
            type="button"
          >
            <i
              className="fas fa-jedi"
            />
          </button>
          <div
            className="info"
          >
            <p>
              <p>
                Vehicle:  wheeled
              </p>
              <p>
                Model:  Digger Crawler
              </p>
              <p>
                Sand Crawler
              </p>
              <p>
                Passengers:  30
              </p>
            </p>
          </div>
        </div>,
      ],
      "className": "cardContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="favorite-btn"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-jedi"
              />
            </button>,
            <div
              className="info"
            >
              <p>
                <p>
                  Vehicle:  wheeled
                </p>
                <p>
                  Model:  Digger Crawler
                </p>
                <p>
                  Sand Crawler
                </p>
                <p>
                  Passengers:  30
                </p>
              </p>
            </div>,
          ],
          "className": "vehicle-card card",
          "id": "no",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-jedi"
              />,
              "className": "favorite-btn",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-jedi",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                <p>
                  Vehicle:  wheeled
                </p>
                <p>
                  Model:  Digger Crawler
                </p>
                <p>
                  Sand Crawler
                </p>
                <p>
                  Passengers:  30
                </p>
              </p>,
              "className": "info",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Vehicle:  wheeled
                  </p>,
                  <p>
                    Model:  Digger Crawler
                  </p>,
                  <p>
                    Sand Crawler
                  </p>,
                  <p>
                    Passengers:  30
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Vehicle:  wheeled",
                  },
                  "ref": null,
                  "rendered": "Vehicle:  wheeled",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Model:  Digger Crawler",
                  },
                  "ref": null,
                  "rendered": "Model:  Digger Crawler",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sand Crawler",
                  },
                  "ref": null,
                  "rendered": "Sand Crawler",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Passengers:  30",
                  },
                  "ref": null,
                  "rendered": "Passengers:  30",
                  "type": "p",
                },
              ],
              "type": "p",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="vehicle-card card"
            id="no"
          >
            <button
              className="favorite-btn"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-jedi"
              />
            </button>
            <div
              className="info"
            >
              <p>
                <p>
                  Vehicle:  wheeled
                </p>
                <p>
                  Model:  Digger Crawler
                </p>
                <p>
                  Sand Crawler
                </p>
                <p>
                  Passengers:  30
                </p>
              </p>
            </div>
          </div>,
        ],
        "className": "cardContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="favorite-btn"
                onClick={[Function]}
                type="button"
              >
                <i
                  className="fas fa-jedi"
                />
              </button>,
              <div
                className="info"
              >
                <p>
                  <p>
                    Vehicle:  wheeled
                  </p>
                  <p>
                    Model:  Digger Crawler
                  </p>
                  <p>
                    Sand Crawler
                  </p>
                  <p>
                    Passengers:  30
                  </p>
                </p>
              </div>,
            ],
            "className": "vehicle-card card",
            "id": "no",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-jedi"
                />,
                "className": "favorite-btn",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-jedi",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  <p>
                    Vehicle:  wheeled
                  </p>
                  <p>
                    Model:  Digger Crawler
                  </p>
                  <p>
                    Sand Crawler
                  </p>
                  <p>
                    Passengers:  30
                  </p>
                </p>,
                "className": "info",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p>
                      Vehicle:  wheeled
                    </p>,
                    <p>
                      Model:  Digger Crawler
                    </p>,
                    <p>
                      Sand Crawler
                    </p>,
                    <p>
                      Passengers:  30
                    </p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Vehicle:  wheeled",
                    },
                    "ref": null,
                    "rendered": "Vehicle:  wheeled",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Model:  Digger Crawler",
                    },
                    "ref": null,
                    "rendered": "Model:  Digger Crawler",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Sand Crawler",
                    },
                    "ref": null,
                    "rendered": "Sand Crawler",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Passengers:  30",
                    },
                    "ref": null,
                    "rendered": "Passengers:  30",
                    "type": "p",
                  },
                ],
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
